client/consensus/manual-seal/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! An instant sealing engine: we always author on every slot, and accept all blocks.
+//! This is suitable for single-node development environments.
+
+use consensus_common::{
+	self, BlockImport, Environment, Proposer, BlockCheckParams,
+	ForkChoiceStrategy, BlockImportParams, BlockOrigin,
+	ImportResult, SelectChain,
+};
+use consensus_common::import_queue::{BasicQueue, CacheKeyId, Verifier, BoxBlockImport};
+use sr_primitives::traits::{Block as BlockT};
+use sr_primitives::Justification;
+use parking_lot::Mutex;
+use futures::prelude::*;
+use transaction_pool::txpool::{self, Pool as TransactionPool};
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+pub mod rpc;
+use rpc::EngineCommand;
+
+/// The synchronous block-import worker of the engine.
+pub struct ManualSealBlockImport<I> {
+	inner: I,
+}
+
+impl<I> From<I> for ManualSealBlockImport<I> {
+	fn from(i: I) -> Self {
+		ManualSealBlockImport { inner: i }
+	}
+}
+
+impl<B: BlockT, I: BlockImport<B>> BlockImport<B> for ManualSealBlockImport<I> {
+	type Error = I::Error;
+
+	fn check_block(&mut self, block: BlockCheckParams<B>) -> Result<ImportResult, Self::Error>
+	{
+		self.inner.check_block(block)
+	}
+
+	fn import_block(
+		&mut self,
+		block: BlockImportParams<B>,
+		cache: HashMap<CacheKeyId, Vec<u8>>,
+	) -> Result<ImportResult, Self::Error> {
+		// TODO: strip out post-digest.
+
+		self.inner.import_block(block, cache)
+	}
+}
+
+/// The verifier for the manual seal engine; instantly finalizes.
```

Comment:
Would probably be good to control finality manually as well.

Reply:
how would this work?

Reply:
via RPC I suppose, so basically you mark the block as final whenever you like. With instant finality there is no way to revert the latest state or build a fork.